server/athenian/api/internal/features/entries.py

```diff
@@ -1772,20 +1769,16 @@ async def _calc_pull_request_facts_github(
             tasks.append(done_jira_map_task)
         done_deps, *done_jira_map = await gather(*tasks)
         if with_jira_map:
-            jira_map = done_jira_map[0]
-            for pr, jira_key in zip(
-                new_jira.get_level_values(0).values, new_jira.get_level_values(1).values,
-            ):
-                jira_map.setdefault(pr, []).append(jira_key)
-            for (node_id, _), f in precomputed_facts.items():
-                try:
-                    f.jira_ids = jira_map[node_id]
-                except KeyError:
-                    continue
+            jira_mappings = done_jira_map[0]
+            mined_jira_mappings = PullRequestJiraMapper.load_from_df(miner.dfs.jiras)
+            for pr_id, mined_jira_mapping in mined_jira_mappings.items():
+                if pr_id not in jira_mappings:
+                    jira_mappings[pr_id] = mined_jira_mapping
+                else:
+                    jira_mappings[pr_id] = jira_mappings[pr_id].merge(mined_jira_mapping)
```

Comment:
This will be a performance bottleneck. Pandas does merges very slowly, we have to rewrite to pure numpy.